### ISSUE TITLE
Popup breaks the positioning of Dialog and ConfirmDialog

### ISSUE THREAD

**Issue by mlindfors** [vaadin-component-factory/popup#1]
With a simple 14.1.16 archetype project and version 2.1.3 of the add-on the following happens:

When opening a Dialog or a ConfirmDialog it should be aligned and justified to the center of the window. For some reason the style setting for the alignment gets overwritten to flex-start if the Maven dependency for Popup add-on is present in the pom.xml of the project.

Steps to reproduce:
- create a new project (like a plain java servlet project from vaadin.com/start)
- add the dependency to the Popup add-on version 2.1.3 into the pom.xml, and update the Maven project
- add the following snippet into the constructor of MainView:
```        Dialog dialog = new Dialog();
        Button openDialog = new Button("Open dialog");
        openDialog.addClickListener(event -> dialog.open());
        add(openDialog);
```
- run the application, and click the "Open dialog" button

Expected outcome: 
the dialog opens centered inside the window

Actual outcome: 
the dialog opens to the left edge of the window

**Comment by TatuLund** [comment]
@gianluca-valentini I investigated the code of the add-on, the problem is a great mystery to me. The implementation is correct. The strange thing is that styles do not leak when using compatibility mode, but this does indeed happen in npm. One possibility is that the issue is not actually in the add-on itself, but some other framework component involved.

**Comment by gianluca-valentini** [comment]
I need a solution because currently the use of this component create problems to **vaadin-date-picker** too.
![image](https://user-images.githubusercontent.com/3224592/79989846-9eb51700-84b0-11ea-989b-ed782dc77745.png)

Or some workaround or a similar compontent that I can use

**Comment by web-padawan** [comment]
The component should be refactored, to be aligned with other Vaadin components.

Especially, [this line](https://github.com/vaadin-component-factory/vcf-popup/blob/9841bbe442cb284401671bafe3dc18b1481d05d4/theme/lumo/vcf-popup-styles.js#L63) is not a proper way of using overlay:

1. popup should have `vaadin-popup-overlay`, which extends `vaadin-overlay` rather than using overlay directly. This would prevent such style leaks to other components (dialog etc)
2. theme should be applied to `vaadin-popup-overlay`

Also, please consider using [registerStyles](https://github.com/vaadin/vaadin-themable-mixin#styling-in-javascript) API instead of creating `dom-module` manually.

**Comment by TatuLund** [comment]
Fixed in version 2.1.4
